tools-local/init-build.proj

```diff
@@ -0,0 +1,323 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="15.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <!-- Don't try to import the target info file when building this project -->
+    <GeneratingStaticPropertiesFile>true</GeneratingStaticPropertiesFile>
+  </PropertyGroup>
+  <Import Project="..\dir.props" />
+
+  <UsingTask AssemblyFile="$(TasksBinDir)Microsoft.DotNet.SourceBuild.Tasks.dll" TaskName="GetHostInformation" />
+  <UsingTask AssemblyFile="$(TasksBinDir)Microsoft.DotNet.SourceBuild.Tasks.dll" TaskName="BuildDummyPackages" />
+
+  <PropertyGroup>
+    <BuildCompetedSuccessSemaphore>$(BaseIntermediatePath)/init-build-proj.complete</BuildCompetedSuccessSemaphore>
+  </PropertyGroup>
+
+  <Target Name="Build"
+    Inputs="$(MSBuildThisFile)"
+    Outputs="$(BuildCompetedSuccessSemaphore)"
+    >
+    <CallTarget Targets="BuildTasks;WriteDynamicPropsToStaticPropsFiles;GenerateRootFs;ApplyPatches;BuildDummyPackages;BuildReferenceAssemblies" />
+    <Touch Files="$(BuildCompetedSuccessSemaphore)" AlwaysCreate="true" />
+  </Target>
+
+  <Target Name="Clean">
+    <Delete Files="$(BuildCompetedSuccessSemaphore)" />
+  </Target>
+
+  <Target Name="BuildTasks">
+    <Exec Command="$(DotNetCliToolDir)dotnet build tasks\Microsoft.DotNet.SourceBuild.Tasks\Microsoft.DotNet.SourceBuild.Tasks.csproj" />
```

Comment:
Does this project need to be restored first?  or are we using a new enough CLI that restore will implicitly happen?

Reply:
From my testing we don't need to restore so I assume we have a new enough CLI.